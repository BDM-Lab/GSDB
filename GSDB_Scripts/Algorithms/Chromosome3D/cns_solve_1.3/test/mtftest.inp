structure

data_cns_mtf

_cns_mtf.title
; FILENAME="/data/rwgk/test/tmp.mtf"
  file toppar/protein.link
  this is a macro to define standard protein peptide bonds
  and termini to generate a protein sequence.
  DATE:18-Nov-98  14:17:41       created by user: rwgk
  VERSION:0.4a
;

loop_
_cns_mtf_atom.id
_cns_mtf_atom.segment_id
_cns_mtf_atom.residue_id
_cns_mtf_atom.residue_name
_cns_mtf_atom.atom_name
_cns_mtf_atom.chemical_type
_cns_mtf_atom.charge
_cns_mtf_atom.atom_mass
1 ' ' '1' 'ASP' 'CB' 'CH2E' -0.160000 14.0270
2 ' ' '1' ' SP' 'CG' 'C' 0.360000 12.0110
3 ' ' '1' '  P' 'OD1' 'OC' -0.600000 15.9994
4 ' ' '1' ' ' 'OD2' 'OC' -0.600000 15.9994
5 ' ' '1' '\pSP' 'C' 'C' 0.550000 12.0110
6 ' ' '1' 'A\pP' 'O' 'O' -0.550000 15.9994
7 ' ' '1' 'AS\p' 'N' 'NH3' -0.300000 14.0067
8 ' ' '1' '\p\p\p' 'CA' 'CH1E' 0.250000 13.0190
9 ' ' '2' 'THR' 'N' 'NH1' -0.350000 14.0067
10 ' ' '2' '\\HR' 'CA' 'CH1E' 0.100000 13.0190
11 ' ' '2' 'T\\R' 'CB' 'CH1E' 0.250000 13.0190
12 ' ' '2' 'TH\\' 'OG1' 'OH1' -0.650000 15.9994
13 ' ' '2' '\\\\\\' 'CG2' 'CH3E' 0.000000E+00 15.0350
14 ' ' '2' 'THR' 'C' 'C' 0.550000 12.0110
15 ' ' '2' 'THR' 'O' 'O' -0.550000 15.9994
16 ' ' '3' 'CYS' 'N' 'NH1' -0.350000 14.0067
17 ' ' '3' 'CYS' 'CA' 'CH1E' 0.100000 13.0190
18 ' ' '3' 'CYS' 'CB' 'CH2E' 0.190000 14.0270
19 ' ' '3' 'CYS' 'SG' 'SH1E' -0.190000 33.0680
20 ' ' '3' 'CYS' 'C' 'C' 0.550000 12.0110
21 ' ' '3' 'CYS' 'O' 'O' -0.550000 15.9994
22 ' ' '4' 'LEU' 'N' 'NH1' -0.350000 14.0067
23 ' ' '4' 'LEU' 'CA' 'CH1E' 0.100000 13.0190
24 ' ' '4' 'LEU' 'CB' 'CH2E' 0.000000E+00 14.0270
25 ' ' '4' 'LEU' 'CG' 'CH1E' 0.000000E+00 13.0190
26 ' ' '4' 'LEU' 'CD1' 'CH3E' 0.000000E+00 15.0350
27 ' ' '4' 'LEU' 'CD2' 'CH3E' 0.000000E+00 15.0350
28 ' ' '4' 'LEU' 'C' 'C' 0.140000 12.0110
29 ' ' '4' 'LEU' 'O' 'OC' -0.570000 15.9994
30 ' ' '4' 'LEU' 'OT' 'OC' -0.570000 15.9994
-1 ' ' ' ' ' ' ' ' ' ' -1.00000 -1.00000

loop_
_cns_mtf_bond.id[1]
_cns_mtf_bond.id[2]
7 8
8 5
5 6
8 1
1 2
2 3
2 4
9 10
10 14
14 15
10 11
11 12
11 13
5 9
16 17
17 20
20 21
17 18
18 19
14 16
22 23
23 28
23 24
24 25
25 26
25 27
20 22
28 29
28 30
-1 -1

loop_
_cns_mtf_angle.id[1]
_cns_mtf_angle.id[2]
_cns_mtf_angle.id[3]
7 8 5
7 8 1
8 5 6
5 8 1
8 1 2
1 2 3
1 2 4
3 2 4
9 10 14
9 10 11
10 14 15
14 10 11
10 11 12
10 11 13
12 11 13
8 5 9
6 5 9
5 9 10
16 17 20
16 17 18
17 20 21
20 17 18
17 18 19
10 14 16
15 14 16
14 16 17
22 23 28
22 23 24
28 23 24
23 24 25
24 25 26
24 25 27
26 25 27
17 20 22
21 20 22
20 22 23
23 28 29
23 28 30
29 28 30
-1 -1 -1

loop_
_cns_mtf_dihedral.id[1]
_cns_mtf_dihedral.id[2]
_cns_mtf_dihedral.id[3]
_cns_mtf_dihedral.id[4]
7 8 1 2
8 1 2 3
9 10 11 12
5 9 10 14
7 8 5 9
8 5 9 10
16 17 18 19
14 16 17 20
9 10 14 16
10 14 16 17
22 23 24 25
23 24 25 27
20 22 23 28
16 17 20 22
17 20 22 23
22 23 28 30
-1 -1 -1 -1

loop_
_cns_mtf_improper.id[1]
_cns_mtf_improper.id[2]
_cns_mtf_improper.id[3]
_cns_mtf_improper.id[4]
8 7 5 1
2 3 4 1
10 9 14 11
11 12 13 10
5 8 9 6
17 16 20 18
14 10 16 15
23 22 28 24
25 27 26 24
20 17 22 21
28 23 30 29
-1 -1 -1 -1

loop_
_cns_mtf_explicit_nonbonded_exclusion.inb
-1

loop_
_cns_mtf_explicit_nonbonded_exclusion.iblo
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
-1

loop_
_cns_mtf_group_linked_list.first_atom_id
0
6
8
15
21
27
-1

end

write structure output=testmtf.dat end

structure reset end
structure @testmtf.dat end
structure

data_cns_mtf

_cns_mtf.title
; FILENAME=""
  DATE:02-Jul-98  15:34:15       created by user: rwgk
  VERSION:0.3r
;

loop_
_cns_mtf_atom.id
_cns_mtf_atom.segment_id
_cns_mtf_atom.residue_id
_cns_mtf_atom.residue_name
_cns_mtf_atom.atom_name
_cns_mtf_atom.chemical_type
_cns_mtf_atom.charge
_cns_mtf_atom.atom_mass
      -1 '    ' '    ' '    ' '    ' '    '   -1.00000      -1.00000    

loop_
_cns_mtf_bond.id[1]
_cns_mtf_bond.id[2]
      -1       -1

loop_
_cns_mtf_angle.id[1]
_cns_mtf_angle.id[2]
_cns_mtf_angle.id[3]
      -1       -1       -1

loop_
_cns_mtf_dihedral.id[1]
_cns_mtf_dihedral.id[2]
_cns_mtf_dihedral.id[3]
_cns_mtf_dihedral.id[4]
      -1       -1       -1       -1

loop_
_cns_mtf_improper.id[1]
_cns_mtf_improper.id[2]
_cns_mtf_improper.id[3]
_cns_mtf_improper.id[4]
      -1       -1       -1       -1

loop_
_cns_mtf_explicit_nonbonded_exclusion.inb
      -1

loop_
_cns_mtf_explicit_nonbonded_exclusion.iblo
      -1

loop_
_cns_mtf_group_linked_list.first_atom_id
      -1

end

show (resname) (all)

structure @testmtf.dat end

write structure end

stop
