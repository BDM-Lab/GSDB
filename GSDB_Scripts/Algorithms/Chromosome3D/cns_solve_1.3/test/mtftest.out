          ============================================================
          |                                                          |
          |            Crystallography & NMR System (CNS)            |
          |                         CNSsolve                         |
          |                                                          |
          ============================================================
           Version: 1.3
           Status: General release
          ============================================================
           Written by: A.T.Brunger, P.D.Adams, G.M.Clore, W.L.DeLano,
                       P.Gros, R.W.Grosse-Kunstleve,J.-S.Jiang,J.M.Krahn,
                       J.Kuszewski, M.Nilges, N.S.Pannu, R.J.Read,
                       L.M.Rice, G.F.Schroeder, T.Simonson, G.L.Warren.
           Copyright (c) 1997-2010 Yale University
          ============================================================
           Running on machine: lewis4-r710-login-node223 (x86_64/Linux,64-bit)
                               with 16 threads
           Program started by: oeow39
           Program started at: 16:08:04 on 05-Jul-2018
          ============================================================

 FFT3C: Using FFTPACK4.1

 CNSsolve>structure 
 STRUcture> 
 STRUcture>data_cns_mtf 
 REMARKS FILENAME="/data/rwgk/test/tmp.mtf"
 REMARKS file toppar/protein.link
 REMARKS this is a macro to define standard protein peptide bonds
 REMARKS and termini to generate a protein sequence.
 REMARKS DATE:18-Nov-98  14:17:41       created by user: rwgk
 REMARKS VERSION:0.4a
 Status of internal molecular topology database:
 -> NATOM=         30(MAXA=     1000000)  NBOND=         29(MAXB=     1000000)
 -> NTHETA=        39(MAXT=     2000000)  NGRP=           6(MAXGRP=   1000000)
 -> NPHI=          16(MAXP=     2000000)  NIMPHI=        11(MAXIMP=   1000000)
 -> NNB=            0(MAXNB=    1000000) 
 STRUcture> 
 STRUcture>end 
 CNSsolve> 
 CNSsolve>write structure output=testmtf.dat end 
 ASSFIL: file testmtf.dat opened.
 CNSsolve> 
 CNSsolve>structure reset end 
 Status of internal molecular topology database:
 -> NATOM=          0(MAXA=     1000000)  NBOND=          0(MAXB=     1000000)
 -> NTHETA=         0(MAXT=     2000000)  NGRP=           0(MAXGRP=   1000000)
 -> NPHI=           0(MAXP=     2000000)  NIMPHI=         0(MAXIMP=   1000000)
 -> NNB=            0(MAXNB=    1000000) 
 CNSsolve>structure @testmtf.dat end 
 ASSFIL: file testmtf.dat opened.
 STRUcture>data_cns_mtf 
 REMARKS FILENAME="testmtf.dat"
 REMARKS file toppar/protein.link
 REMARKS this is a macro to define standard protein peptide bonds
 REMARKS and termini to generate a protein sequence.
 REMARKS DATE:05-Jul-2018  16:08:04       created by user: oeow39
 REMARKS VERSION:1.3
 Status of internal molecular topology database:
 -> NATOM=         30(MAXA=     1000000)  NBOND=         29(MAXB=     1000000)
 -> NTHETA=        39(MAXT=     2000000)  NGRP=           6(MAXGRP=   1000000)
 -> NPHI=          16(MAXP=     2000000)  NIMPHI=        11(MAXIMP=   1000000)
 -> NNB=            0(MAXNB=    1000000) 
 STRUcture> 
 STRUcture> end 
 CNSsolve>structure 
 STRUcture> 
 STRUcture>data_cns_mtf 
 REMARKS FILENAME=""
 REMARKS DATE:02-Jul-98  15:34:15       created by user: rwgk
 REMARKS VERSION:0.3r
 Status of internal molecular topology database:
 -> NATOM=         30(MAXA=     1000000)  NBOND=         29(MAXB=     1000000)
 -> NTHETA=        39(MAXT=     2000000)  NGRP=           6(MAXGRP=   1000000)
 -> NPHI=          16(MAXP=     2000000)  NIMPHI=        11(MAXIMP=   1000000)
 -> NNB=            0(MAXNB=    1000000) 
 STRUcture> 
 STRUcture>end 
 CNSsolve> 
 CNSsolve>show (resname) (all) 
 SELRPN:     30 atoms have been selected out of     30
 (      ASP  1    CB   )  ASP 
 (       SP  1    CG   )   SP 
 (        P  1    OD1  )    P 
 (           1    OD2  )      
 (      'SP  1    C    )  'SP 
 (      A'P  1    O    )  A'P 
 (      AS'  1    N    )  AS' 
 (      '''  1    CA   )  ''' 
 (      THR  2    N    )  THR 
 (      \HR  2    CA   )  \HR 
 (      T\R  2    CB   )  T\R 
 (      TH\  2    OG1  )  TH\ 
 (      \\\  2    CG2  )  \\\ 
 (      THR  2    C    )  THR 
 (      THR  2    O    )  THR 
 (      CYS  3    N    )  CYS 
 (      CYS  3    CA   )  CYS 
 (      CYS  3    CB   )  CYS 
 (      CYS  3    SG   )  CYS 
 (      CYS  3    C    )  CYS 
 (      CYS  3    O    )  CYS 
 (      LEU  4    N    )  LEU 
 (      LEU  4    CA   )  LEU 
 (      LEU  4    CB   )  LEU 
 (      LEU  4    CG   )  LEU 
 (      LEU  4    CD1  )  LEU 
 (      LEU  4    CD2  )  LEU 
 (      LEU  4    C    )  LEU 
 (      LEU  4    O    )  LEU 
 (      LEU  4    OT   )  LEU 
 CNSsolve> 
 CNSsolve>structure @testmtf.dat end 
 ASSFIL: file testmtf.dat opened.
 STRUcture>data_cns_mtf 
 REMARKS FILENAME="testmtf.dat"
 REMARKS file toppar/protein.link
 REMARKS this is a macro to define standard protein peptide bonds
 REMARKS and termini to generate a protein sequence.
 REMARKS DATE:05-Jul-2018  16:08:04       created by user: oeow39
 REMARKS VERSION:1.3
 Status of internal molecular topology database:
 -> NATOM=         60(MAXA=     1000000)  NBOND=         58(MAXB=     1000000)
 -> NTHETA=        78(MAXT=     2000000)  NGRP=          12(MAXGRP=   1000000)
 -> NPHI=          32(MAXP=     2000000)  NIMPHI=        22(MAXIMP=   1000000)
 -> NNB=            0(MAXNB=    1000000) 
 STRUcture> 
 STRUcture> end 
 CNSsolve> 
 CNSsolve>write structure end 
data_cns_mtf

_cns_mtf.title
; FILENAME="OUTPUT"
  file toppar/protein.link
  this is a macro to define standard protein peptide bonds
  and termini to generate a protein sequence.
  DATE:05-Jul-2018  16:08:04       created by user: oeow39
  VERSION:1.3
;

loop_
_cns_mtf_atom.id
_cns_mtf_atom.segment_id
_cns_mtf_atom.residue_id
_cns_mtf_atom.residue_name
_cns_mtf_atom.atom_name
_cns_mtf_atom.chemical_type
_cns_mtf_atom.charge
_cns_mtf_atom.atom_mass
1 ' ' '1' 'ASP' 'CB' 'CH2E' -0.160000 14.0270
2 ' ' '1' ' SP' 'CG' 'C' 0.360000 12.0110
3 ' ' '1' '  P' 'OD1' 'OC' -0.600000 15.9994
4 ' ' '1' ' ' 'OD2' 'OC' -0.600000 15.9994
5 ' ' '1' '\pSP' 'C' 'C' 0.550000 12.0110
6 ' ' '1' 'A\pP' 'O' 'O' -0.550000 15.9994
7 ' ' '1' 'AS\p' 'N' 'NH3' -0.300000 14.0067
8 ' ' '1' '\p\p\p' 'CA' 'CH1E' 0.250000 13.0190
9 ' ' '2' 'THR' 'N' 'NH1' -0.350000 14.0067
10 ' ' '2' '\\HR' 'CA' 'CH1E' 0.100000 13.0190
11 ' ' '2' 'T\\R' 'CB' 'CH1E' 0.250000 13.0190
12 ' ' '2' 'TH\\' 'OG1' 'OH1' -0.650000 15.9994
13 ' ' '2' '\\\\\\' 'CG2' 'CH3E' 0.00000 15.0350
14 ' ' '2' 'THR' 'C' 'C' 0.550000 12.0110
15 ' ' '2' 'THR' 'O' 'O' -0.550000 15.9994
16 ' ' '3' 'CYS' 'N' 'NH1' -0.350000 14.0067
17 ' ' '3' 'CYS' 'CA' 'CH1E' 0.100000 13.0190
18 ' ' '3' 'CYS' 'CB' 'CH2E' 0.190000 14.0270
19 ' ' '3' 'CYS' 'SG' 'SH1E' -0.190000 33.0680
20 ' ' '3' 'CYS' 'C' 'C' 0.550000 12.0110
21 ' ' '3' 'CYS' 'O' 'O' -0.550000 15.9994
22 ' ' '4' 'LEU' 'N' 'NH1' -0.350000 14.0067
23 ' ' '4' 'LEU' 'CA' 'CH1E' 0.100000 13.0190
24 ' ' '4' 'LEU' 'CB' 'CH2E' 0.00000 14.0270
25 ' ' '4' 'LEU' 'CG' 'CH1E' 0.00000 13.0190
26 ' ' '4' 'LEU' 'CD1' 'CH3E' 0.00000 15.0350
27 ' ' '4' 'LEU' 'CD2' 'CH3E' 0.00000 15.0350
28 ' ' '4' 'LEU' 'C' 'C' 0.140000 12.0110
29 ' ' '4' 'LEU' 'O' 'OC' -0.570000 15.9994
30 ' ' '4' 'LEU' 'OT' 'OC' -0.570000 15.9994
31 ' ' '1' 'ASP' 'CB' 'CH2E' -0.160000 14.0270
32 ' ' '1' ' SP' 'CG' 'C' 0.360000 12.0110
33 ' ' '1' '  P' 'OD1' 'OC' -0.600000 15.9994
34 ' ' '1' ' ' 'OD2' 'OC' -0.600000 15.9994
35 ' ' '1' '\pSP' 'C' 'C' 0.550000 12.0110
36 ' ' '1' 'A\pP' 'O' 'O' -0.550000 15.9994
37 ' ' '1' 'AS\p' 'N' 'NH3' -0.300000 14.0067
38 ' ' '1' '\p\p\p' 'CA' 'CH1E' 0.250000 13.0190
39 ' ' '2' 'THR' 'N' 'NH1' -0.350000 14.0067
40 ' ' '2' '\\HR' 'CA' 'CH1E' 0.100000 13.0190
41 ' ' '2' 'T\\R' 'CB' 'CH1E' 0.250000 13.0190
42 ' ' '2' 'TH\\' 'OG1' 'OH1' -0.650000 15.9994
43 ' ' '2' '\\\\\\' 'CG2' 'CH3E' 0.00000 15.0350
44 ' ' '2' 'THR' 'C' 'C' 0.550000 12.0110
45 ' ' '2' 'THR' 'O' 'O' -0.550000 15.9994
46 ' ' '3' 'CYS' 'N' 'NH1' -0.350000 14.0067
47 ' ' '3' 'CYS' 'CA' 'CH1E' 0.100000 13.0190
48 ' ' '3' 'CYS' 'CB' 'CH2E' 0.190000 14.0270
49 ' ' '3' 'CYS' 'SG' 'SH1E' -0.190000 33.0680
50 ' ' '3' 'CYS' 'C' 'C' 0.550000 12.0110
51 ' ' '3' 'CYS' 'O' 'O' -0.550000 15.9994
52 ' ' '4' 'LEU' 'N' 'NH1' -0.350000 14.0067
53 ' ' '4' 'LEU' 'CA' 'CH1E' 0.100000 13.0190
54 ' ' '4' 'LEU' 'CB' 'CH2E' 0.00000 14.0270
55 ' ' '4' 'LEU' 'CG' 'CH1E' 0.00000 13.0190
56 ' ' '4' 'LEU' 'CD1' 'CH3E' 0.00000 15.0350
57 ' ' '4' 'LEU' 'CD2' 'CH3E' 0.00000 15.0350
58 ' ' '4' 'LEU' 'C' 'C' 0.140000 12.0110
59 ' ' '4' 'LEU' 'O' 'OC' -0.570000 15.9994
60 ' ' '4' 'LEU' 'OT' 'OC' -0.570000 15.9994
-1 ' ' ' ' ' ' ' ' ' ' -1.00000 -1.00000

loop_
_cns_mtf_bond.id[1]
_cns_mtf_bond.id[2]
7 8
8 5
5 6
8 1
1 2
2 3
2 4
9 10
10 14
14 15
10 11
11 12
11 13
5 9
16 17
17 20
20 21
17 18
18 19
14 16
22 23
23 28
23 24
24 25
25 26
25 27
20 22
28 29
28 30
37 38
38 35
35 36
38 31
31 32
32 33
32 34
39 40
40 44
44 45
40 41
41 42
41 43
35 39
46 47
47 50
50 51
47 48
48 49
44 46
52 53
53 58
53 54
54 55
55 56
55 57
50 52
58 59
58 60
-1 -1

loop_
_cns_mtf_angle.id[1]
_cns_mtf_angle.id[2]
_cns_mtf_angle.id[3]
7 8 5
7 8 1
8 5 6
5 8 1
8 1 2
1 2 3
1 2 4
3 2 4
9 10 14
9 10 11
10 14 15
14 10 11
10 11 12
10 11 13
12 11 13
8 5 9
6 5 9
5 9 10
16 17 20
16 17 18
17 20 21
20 17 18
17 18 19
10 14 16
15 14 16
14 16 17
22 23 28
22 23 24
28 23 24
23 24 25
24 25 26
24 25 27
26 25 27
17 20 22
21 20 22
20 22 23
23 28 29
23 28 30
29 28 30
37 38 35
37 38 31
38 35 36
35 38 31
38 31 32
31 32 33
31 32 34
33 32 34
39 40 44
39 40 41
40 44 45
44 40 41
40 41 42
40 41 43
42 41 43
38 35 39
36 35 39
35 39 40
46 47 50
46 47 48
47 50 51
50 47 48
47 48 49
40 44 46
45 44 46
44 46 47
52 53 58
52 53 54
58 53 54
53 54 55
54 55 56
54 55 57
56 55 57
47 50 52
51 50 52
50 52 53
53 58 59
53 58 60
59 58 60
-1 -1 -1

loop_
_cns_mtf_dihedral.id[1]
_cns_mtf_dihedral.id[2]
_cns_mtf_dihedral.id[3]
_cns_mtf_dihedral.id[4]
7 8 1 2
8 1 2 3
9 10 11 12
5 9 10 14
7 8 5 9
8 5 9 10
16 17 18 19
14 16 17 20
9 10 14 16
10 14 16 17
22 23 24 25
23 24 25 27
20 22 23 28
16 17 20 22
17 20 22 23
22 23 28 30
37 38 31 32
38 31 32 33
39 40 41 42
35 39 40 44
37 38 35 39
38 35 39 40
46 47 48 49
44 46 47 50
39 40 44 46
40 44 46 47
52 53 54 55
53 54 55 57
50 52 53 58
46 47 50 52
47 50 52 53
52 53 58 60
-1 -1 -1 -1

loop_
_cns_mtf_improper.id[1]
_cns_mtf_improper.id[2]
_cns_mtf_improper.id[3]
_cns_mtf_improper.id[4]
8 7 5 1
2 3 4 1
10 9 14 11
11 12 13 10
5 8 9 6
17 16 20 18
14 10 16 15
23 22 28 24
25 27 26 24
20 17 22 21
28 23 30 29
38 37 35 31
32 33 34 31
40 39 44 41
41 42 43 40
35 38 39 36
47 46 50 48
44 40 46 45
53 52 58 54
55 57 56 54
50 47 52 51
58 53 60 59
-1 -1 -1 -1

loop_
_cns_mtf_explicit_nonbonded_exclusion.inb
-1

loop_
_cns_mtf_explicit_nonbonded_exclusion.iblo
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
-1

loop_
_cns_mtf_group_linked_list.first_atom_id
0
6
8
15
21
27
30
36
38
45
51
57
-1

 CNSsolve> 
 CNSsolve>stop 
          ============================================================
           Maximum dynamic memory allocation:     8201488 bytes
           Maximum dynamic memory overhead:           208 bytes
           Program started at: 16:08:04 on 05-Jul-2018
           Program stopped at: 16:08:04 on 05-Jul-2018
           CPU time used:       0.0173 seconds
          ============================================================
